src/resource_reader.rs

```diff
@@ -11,16 +15,34 @@ pub struct FileRead {
     pub path: String,
 }
 
+lazy_static! {
+    // Global cache accessible by all instances of ResourceReader
+    static ref CACHE: Mutex<HashMap<String, String>> = Mutex::new(HashMap::new());
+}
+
 pub struct ResourceReader {
     runtime: TargetRuntime,
+    enable_cache: bool,
```

Comment:
Keep the cache local to the ResourceReader instead of a static cache.

Reply:
I tried that, but it won't help with the duplication issue, since the reading is done from two separate instances of ResourceReader.